pkg/endpointmanager/manager.go

```diff
@@ -653,6 +653,13 @@ func (mgr *endpointManager) AddEndpoint(owner regeneration.Owner, ep *endpoint.E
 		return fmt.Errorf("Endpoint ID is already set to %d", ep.ID)
 	}
 
+	oldEP := mgr.LookupCEPName(ep.GetK8sNamespaceAndPodName())
+	if oldEP != nil {
+		// This is a container recreation event for an existing Pod.
+		// We want to merge this new endpoint with the existing endpoint.
+		ep.ID = oldEP.ID
+	}
```

Comment:
Is that all that's really needed to "merge" the new endpoint with the existing?

Reply:
Based on my reproduction scenario, yes this replaces the endpoint controller with the new endpoint associated with the CNI ADD. The controller name change along was not sufficient.

Reply:
My suspicion is only setting the ID probably leads to memory leaks due to the previous Endpoint references not being cleaned up/replaced everywhere. Have you looked into that?

Reply:
I agree with Chris on this. We are basically re-using an existing endpoint ID without waiting for all the instances of the "old" endpoint to be removed. I think this might cause some unknown issues. Is this change really necessary? The `EndpointSyncControllerName(name string)` would make the controller "unique" for the pod name, isn't that enough?

Reply:
That was my first approach, but a second controller was still being spawned for the second create. This change deduped the controller, but is running afoul of the docker pod creation during the integration tests.

Let me try removing this again and track down an alternative that doesn't reuse the endpoint ID.